src/syrupy/extensions/amber/serializer.py

```diff
@@ -348,6 +349,17 @@ def serialize_unknown(cls, data: Any, *, depth: int = 0, **kwargs: Any) -> str:
             **kwargs,
         )
 
+    @classmethod
+    def object_attrs(cls, data: Any) -> "Iterable[str]":
+        return (name for name in dir(data) if not name.startswith("_"))
+
+    @classmethod
+    def bypass_custom_repr(cls, data: Any) -> "tuple[Any, ...]":
+        attr_names = list(cls.object_attrs(data))
+        return collections.namedtuple(data.__class__.__name__, attr_names)(
+            **{prop: getattr(data, prop) for prop in attr_names}
+        )
```

Comment:
I was thinking about how the namedtuple serialization is exactly what we want when it came to https://github.com/tophat/syrupy/issues/784.

Regarding the name of the method, this does not really "bypass" custom repr, it's just a utility for constructing a named tuple based on an object's attrs. If it's "bypassing" some serialization, I'd expect it to somehow "communicate" with the serializer itself. I.e. a model where it sets a marker on an object so that the serializer knows to handle it differently.

Instead, I'd rename this to more explicitly be a utility that constructs a named tuple from a custom object. Perhaps `def make_named_tuple` or `def create_named_tuple` / `def convert_obj_to_named_tuple`.

If the idea is to use this in conjunction with the replacer to provide a more elegant solution to the msgspec serializer issue, we'd want a mechanism to apply the replacer to all snapshots. It gets verbose having to specify the replacer per use.

Another way I was thinking of approaching the problem in the long run was to expose some static "registry" / map of custom class types -> serializers to register with the class / or to override.

e.g.

```
class Serializer(AmberDataSerializer):
    custom_classes = { msgspec.Struct: some_serializer }

Reply:
Changed the name to `object_as_named_tuple_class`.

> It gets verbose having to specify the replacer per use.

True, the `replacer` is a low hanging fruit that allows both functionality I wanted to add here. I like the `custom_classes` idea, it could even be used in the `__matcher` assertion property.

Another option would be to allow a change to the `_matcher` property that won't be reverted after each assertion.

```py
@pytest.fixture
def snapshot_bypass_my_custom_repr(snapshot):
  snapshot._matcher = path_type(
    types=(MyCustomReprClass,),
    replacer=lambda data, _: AmberDataSerializer.object_as_named_tuple(data),
  )
  return snapshot

def test_snapshot_object_as_named_tuple_class(snapshot_bypass_my_custom_repr):
    assert MyCustomReprClass() == snapshot_bypass_my_custom_repr(
        exclude=props("prop1"),
    )
```